app/services/search_service.rb

```diff
@@ -50,7 +50,12 @@ def textSearch_all(search_criteria)
   # If no sort_by value present, then it returns a list of profiles ordered by id DESC
   # a recent activity may be a node creation or a node revision
   def profiles(search_criteria)
-    user_scope = SrchScope.find_users(search_criteria.query, limit = 10)
+    user_scope =
+      if search_criteria.field == "username"
+        SrchScope.find_by_username(search_criteria.query, limit = 10)
```

Comment:
Here we might think of setting an optional limit parameter in profiles(search_criteria, limit = 10) and then we could override it if needed?